kratos/processes/levelset_convection_process.h

```diff
@@ -225,19 +248,76 @@ class LevelSetConvectionProcess
             const double Nnew_before = 1.0 - Nold_before;
 
             // Emulate clone time step by copying the new distance onto the old one
-            #pragma omp parallel for
-            for (int i_node = 0; i_node < static_cast<int>(mpDistanceModelPart->NumberOfNodes()); ++i_node){
-                auto it_node = mpDistanceModelPart->NodesBegin() + i_node;
+            if (mBfeccOrder == 0){
```

Comment:
By using an auxiliary internal function you will avoid all these `mBfeccOrder` if statements while avoiding circumventing the `Process` API.

Reply:
Could you please give me an example? You mean adding a separate function to do BFECC convection?

Reply:
In pseudocode, something like:
```c++
if (mDoBFECC) {
    DoStandardConvection() // This is an auxiliary private function with the old implementation
} else {
    DoNewConvection() // This is an auxiliary private function with your enhanced convection
}
```
If some parts of the code are identical in the `DoStandardConvection` and `DoNewConvection` we can create more auxiliary private methods (as a rule of thumb take that a function shouldn't be longer than 100-150 lines :sweat_smile: )